controllers/admin.py

```diff
@@ -719,23 +719,31 @@ def deletecourse():
                 shutil.rmtree(path.join('applications', request.application, 'custom_courses', course_name))
                 session.clear()
             except:
-                response.flash = 'Error, %s does not appear to exist' % course_name
+                session.flash = 'Error, %s does not appear to exist' % course_name
         else:
-            response.flash = 'You are not the instructor of %s' % course_name
+            session.flash = 'You are not the instructor of %s' % course_name
     else:
-        response.flash = 'course, %s, not found' % course_name
+        session.flash = 'course, %s, not found' % course_name
 
     redirect(URL('default','index'))
 
 
 @auth.requires(lambda: verifyInstructorStatus(auth.user.course_name, auth.user), requires_login=True)
 def removeassign():
+    response.headers['content-type'] = 'application/json'
     try:
-        assignment_id = int(request.args[0])
+        assignment_id = int(request.vars['assignid'])
     except:
-        session.flash = "Cannot remove assignment with id of {}".format(request.args[0])
-        return
-    db(db.assignments.id == assignment_id).delete()
+        session.flash = "Cannot remove assignment with id of {}".format(request.vars['assignid'])
```

Comment:
I'm confused. Is this a user-facing endpoint, or something hit by JS? If the first, then return "Success" makes no sense; if the second, this makes no sense.

Reply:
It is called from Javascript, but I added the session.flash when I was testing by calling from the browser with a Naked URL.  -- Hey at least it doesn't just crash anymore!!

Reply:
Progress! IMHO, I'd `return "Cannot remove..."` to be consistent.